File: refactoring/extract_function/extract_function.go
package extract_function

import "fmt"

type OutStanding struct {
	customer string
	amount   int64
}

func PrintOwing() {
	customer := OutStanding{
		customer: "john",
		amount:   100,
	}
	fmt.Print(customer.amount)
	fmt.Print(customer.customer)
}

func NewPrintOwing() {
	customer := OutStanding{
		customer: "john",
		amount:   100,
	}
	printDetail(customer)
}

func printDetail(customer OutStanding) {
	fmt.Print(customer.amount)
	fmt.Print(customer.customer)
}

func main() {
	PrintOwing()
	NewPrintOwing()
}

File: refactoring/extract_function/extract_function.py
# old code
def print_owing(invoice):
    out_standing = {"customer": "john", "amount": 100}

    print(out_standing.get("customer"))
    print(out_standing.get('account'))


# new code
def new_print_owing():
    out_standing = {"customer": "john", "amount": 100}
    print_detail(out_standing)


def print_detail(out_standing):
    print(out_standing.get("customer"))
    print(out_standing.get('account'))
